### ISSUE TITLE
[CLI]: RuntimeError: "histogram_cpu" not implemented for 'BFloat16'

### ISSUE THREAD

**Issue by andstor** [wandb/wandb#3332]
### Describe the bug

When using `BFloat16` floating point format, I get the following runtime error:
```shell
It seems that "histogram_cpu" is not implemented for 'BFloat16' in PyTorch.
```
Offending line:
https://github.com/wandb/client/blob/48b740a930648ce0a7c0f789a4e11042cb24d60b/wandb/wandb_torch.py#L215

Traceback:
```shell
Traceback (most recent call last):
  File "./examples/pytorch/language-modeling/run_clm.py", line 545, in <module>
    main()
  File "./examples/pytorch/language-modeling/run_clm.py", line 493, in main
    train_result = trainer.train(resume_from_checkpoint=checkpoint)
  File "/cluster/home/andstorh/transformers/.venv/lib/python3.8/site-packages/transformers/trainer.py", line 1400, in train
    tr_loss_step = self.training_step(model, inputs)
  File "/cluster/home/andstorh/transformers/.venv/lib/python3.8/site-packages/transformers/trainer.py", line 1984, in training_step
    loss = self.compute_loss(model, inputs)
  File "/cluster/home/andstorh/transformers/.venv/lib/python3.8/site-packages/transformers/trainer.py", line 2016, in compute_loss
    outputs = model(**inputs)
  File "/cluster/home/andstorh/transformers/.venv/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/cluster/home/andstorh/transformers/.venv/lib/python3.8/site-packages/deepspeed/runtime/engine.py", line 1588, in forward
    loss = self.module(*inputs, **kwargs)
  File "/cluster/home/andstorh/transformers/.venv/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1123, in _call_impl
    hook_result = hook(self, input, result)
  File "/cluster/home/andstorh/transformers/.venv/lib/python3.8/site-packages/wandb/wandb_torch.py", line 113, in <lambda>
    lambda mod, inp, outp: parameter_log_hook(
  File "/cluster/home/andstorh/transformers/.venv/lib/python3.8/site-packages/wandb/wandb_torch.py", line 109, in parameter_log_hook
    self.log_tensor_stats(data.cpu(), "parameters/" + prefix + name)
  File "/cluster/home/andstorh/transformers/.venv/lib/python3.8/site-packages/wandb/wandb_torch.py", line 215, in log_tensor_stats
    tensor = flat.histc(bins=self._num_bins, min=tmin, max=tmax)
RuntimeError: "histogram_cpu" not implemented for 'BFloat16'
```

### Environment

WandB version: 0.12.11

OS: Linux-4.18.0-193.el8.x86_64-x86_64-with-glibc2.10

Python version: 3.8.3

Versions of relevant libraries: torch 1.10.2+cu113, transformers: 4.18.0.dev0, and deepspeed: 0.5.10

**Comment by ramit-wandb** [comment]
Hi @andstor,

Thanks for writing in! Would it be possible for you to share a minimal reproduction of this bug? I'll pass it on to our engineering team to have this resolved.

Thanks,
Ramit
Weights and Biases Support

**Comment by andstor** [comment]
Hi @ramit-wandb,

following is a minimal reproduction of the bug.

### Information
Model: GPT-2
Running on GPU cluster with 2 x NVIDIA A100 40G

### To reproduce

Run causal language modeling example training script from Hugging Face: [`transformers/examples/pytorch/language-modeling/run_clm.py.`](https://github.com/huggingface/transformers/blob/master/examples/pytorch/language-modeling/run_clm.py)

Hugging Face launch script:
```bash
export WANDB_LOG_MODEL=true
export WANDB_WATCH=true

deepspeed examples/pytorch/language-modeling/run_clm.py \
    --deepspeed ./zero2_min.json \
    --model_name_or_path gpt2 \
    --dataset_name wikitext \
    --dataset_config_name wikitext-2-raw-v1 \
    --overwrite_output_dir \
    --per_device_train_batch_size 8 \
    --logging_steps 1 \
    --do_train \
    --bf16 \
    --output_dir /tmp/test-clm
```

DeepSpeed config:
```json
{
    "fp16": {
        "enabled": false
    },
    "bfloat16": {
        "enabled": true
    },
    "zero_optimization": {
        "stage": 2,
        "offload_optimizer": {
            "device": "cpu",
            "pin_memory": true
        },
        "allgather_partitions": true,
        "allgather_bucket_size": 2e8,
        "reduce_scatter": true,
        "reduce_bucket_size": 2e8,
        "overlap_comm": true,
        "contiguous_gradients": true
    },
    "gradient_accumulation_steps": "auto",
    "gradient_clipping": "auto",
    "steps_per_print": 100,
    "train_batch_size": "auto",
    "train_micro_batch_size_per_gpu": "auto",
    "wall_clock_breakdown": false
}
```

After 100 steps, it fails with the following error: 
```shell
RuntimeError: "histogram_cpu" not implemented for 'BFloat16'
```

### Debug logs
[debug.log](https://github.com/wandb/client/files/8202222/debug.log)
[debug-internal.log](https://github.com/wandb/client/files/8202221/debug-internal.log)

**Comment by ramit-wandb** [comment]
Hi @andstor,

Thanks for the information! I looked into this further, and it looks like `histogram_cpu` is actually a function belonging to PyTorch, not `wandb`: 

https://github.com/pytorch/pytorch/blob/bc03c1d000dcb8417fc771a49165c2b963278092/aten/src/ATen/native/Histogram.cpp#L345-L350

This is something that would have to be resolved by the PyTorch team. I recommend contacting the PyTorch team to resolve this.

Thanks,
Ramit

**Comment by ramit-wandb** [comment]
Closing this ticket since there is nothing to be done here.

Thanks,
Ramit

**Comment by jncasey** [comment]
This seems to be fixed with the release of pytorch 1.12
